### PR TITLE
Add acceptance test for metrics merging

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR (by Nitya):
- Add an acceptance test to verify metrics merging works for service metrics that support the same

How I've tested this PR: Run the acceptance test and things looked phenomenal (much like Nitya's code)

How I expect reviewers to test this PR: However Nitya deems fit. Probably watch the pipelines go green and Code Review the rest.

Checklist:
- [ ] Bats tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: test/acceptance/tests/metrics/metrics_test.go [hunk 1]
```diff
@@ -81,6 +82,56 @@ func TestMetrics(t *testing.T) {
...
+func TestAppMetrics(t *testing.T) {
+	env := suite.Environment()
+	cfg := suite.Config()
+	ctx := env.DefaultContext(t)
+	ns := ctx.KubectlOptions(t).Namespace
+
+	helmValues := map[string]string{
+		"global.datacenter":                 "dc1",
+		"global.metrics.enabled":            "true",
+		"global.metrics.enableAgentMetrics": "true",
```
Comment (ishustava) [inline test/acceptance/tests/metrics/metrics_test.go:96]: do we need agent metrics for this test?

File: test/acceptance/tests/metrics/metrics_test.go [hunk 2]
```diff
@@ -81,6 +82,56 @@ func TestMetrics(t *testing.T) {
...
+
+		// These are images that are required until https://github.com/hashicorp/consul/pull/9768 and https://github.com/hashicorp/consul-k8s/pull/443
+		"global.imageK8S": "gcr.io/nitya-293720/consul-k8s-dev:metrics30",
+		"global.image":    "gcr.io/nitya-293720/consul-dev:metrics10",
+
+		"connectInject.enabled":                      "true",
+		"connectInject.metrics.defaultEnableMerging": "true",
+
+		"controller.enabled": "true",
```
Comment (ishustava) [inline test/acceptance/tests/metrics/metrics_test.go:105]: do we need controller?

File: test/acceptance/tests/metrics/metrics_test.go [hunk 3]
```diff
@@ -81,6 +82,56 @@ func TestMetrics(t *testing.T) {
...
+
+	// Merged App Metrics
+	podList, err := ctx.KubernetesClient(t).CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{LabelSelector: "app=static-metrics-app"})
+	require.NoError(t, err)
+	require.Len(t, podList.Items, 1)
+	podIP := podList.Items[0].Status.PodIP
+	metricsOutput, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "exec", "deploy/"+staticClientName, "--", "curl", fmt.Sprintf("http://%s:20200/metrics", podIP))
+	require.NoError(t, err)
+	require.Contains(t, metricsOutput, `envoy_cluster_assignment_stale{local_cluster="server",consul_source_service="server",consul_source_namespace="default",consul_source_datacenter="dc1",envoy_cluster_name="local_agent"} 0`)
+	require.Contains(t, metricsOutput, `service_started_total 1`)
```
Comment (ishustava) [inline test/acceptance/tests/metrics/metrics_test.go:132]: could we add a comment here that this is the metric emitted by the app?

**Review by ishustava — APPROVED** [review]
Looks great!

File: .circleci/config.yml [hunk 4]
```diff
@@ -144,7 +144,7 @@ jobs:
                     -kubecontext="kind-dc1" \
                     -secondary-kubecontext="kind-dc2" \
                     -debug-directory="$TEST_RESULTS/debug" \
-                    -consul-k8s-image=docker.mirror.hashicorp.services/hashicorpdev/consul-k8s:latest
+                    -consul-k8s-image=gcr.io/nitya-293720/consul-k8s-dev:metrics30
```
Comment (ndhanushkodi) [inline .circleci/config.yml:147]: On the feature metrics branch, should we use this image so we can see acceptance tests passing once we merge this and #834 in? (I was also thinking about this from connect service metrics pr ( #834 )).

**Review by ndhanushkodi — APPROVED** [review]
Looks great, just thinking about we might do for the k8s image when we merge this and #834 into feature-metrics.

**Comment by ndhanushkodi** [comment]
I can help to resolve the conflicts, I had made some updates to logic in connect-svc-metrics.

File: .circleci/config.yml (hunk 4 above)
Comment (thisisnotashwin) [reply .circleci/config.yml:147]: That makes sense 👍

Comment (thisisnotashwin) [reply .circleci/config.yml:147]: Let's merge this on #834 into the feature branch and then create a potential PR that has the latest images.

**Pull request merged**